homeassistant/config_entries.py

```diff
@@ -1101,12 +1113,12 @@ async def _async_init_reconfigure(
                 return
             await hass.config_entries.flow.async_init(
                 self.domain,
-                context={
```

Comment:
Not sure how hot this is, but in  other places where we have hot paths we create the dict with a type instead of using the callable since it’s a bit faster

ie

context: type = {…}

Reply:
This particular case is inside `_async_init_reconfigure` which can't possibly be a hot path.
The comment is for sure relevant though 👍 

Considering the changes in this PR, I think this case should just create a type annotated dict: https://github.com/home-assistant/core/pull/126291/files#diff-c446f4035d7591f034bfcada83e64b8fc59aff6feb12cbb9d5663dbd83f4e7e6R543-R545

Reply:
Yeah `_async_init_reconfigure` definitely isn't hot.  Its only going to be the discovery flows